python/tk_multi_workfiles/entity_proxy_model.py

```diff
@@ -25,6 +25,42 @@ def __init__(self, parent, compare_sg_fields=None):
         HierarchicalFilteringProxyModel.__init__(self, parent)
         self._compare_fields = compare_sg_fields
 
+    def __ensure_subtree_loaded(self, model_index):
+        """
+        Recursively descend and ensure all data is
+        fetched for the given model index.
+
+        :param model_index: Model index to process
+        """
+        model = model_index.model()
+        if model.canFetchMore(model_index):
+            model.fetchMore(model_index)
+        for child_index in range(model.rowCount(model_index)):
```

Comment:
I'm probably forgetting something here, but if `fetchMore` is asynchronous, won't `rowCount` be unreliable?